src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -180,19 +185,62 @@ public ItemStack getStack(Random rand, int fortune)
     }
     static final List<SeedEntry> seedList = new ArrayList<SeedEntry>();
 
+    /**
+     * A hacked LootPool implementation that makes sure that the old seedList are still supported
+     * for sake of backward compatibilities, and that Forge can still draw items from that list
+     * using the old logic as well. This implementation is NOT for public uses! Modders should
+     * directly use vanilla LootTable instead.
+     */
+    // TODO Remove this when MinecraftForge#addGrassSeed are removed
+    static final class SeedListPool extends LootPool
+    {
+        SeedListPool()
+        {
+            super(new LootEntry[0], new LootCondition[0], new RandomValueRange(0F), new RandomValueRange(0F), "forge:legacy_seed_list");
+        }
+
+        @Override
+        public void generateLoot(Collection<ItemStack> stacks, Random rand, LootContext context)
+        {
+            //Some bad mods hack in and empty our list, so lets not hard crash -.-
+            if (!ForgeHooks.seedList.isEmpty())
+            {
+                SeedEntry entry = WeightedRandom.getRandomItem(rand, ForgeHooks.seedList);
+                if (entry != null)
+                {
+                    // TODO Use custom LootContext for fortune level?
+                    ItemStack result = entry.getStack(rand, (int) context.getLuck());
+                    if (!result.isEmpty())
+                    {
+                        stacks.add(result);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Path of dummy loot table used by populating grass seed drop. The actual content is injected via
+     * {@link ForgeInternalHandler#onLootTableLoad}.
+     */
+    static final ResourceLocation GRASS_SEED_TABLE = new ResourceLocation(ForgeVersion.MOD_ID, "grass_seed");
+
     @Nonnull
     public static ItemStack getGrassSeed(Random rand, int fortune)
     {
-        if (seedList.size() == 0)
+        // Passes fortune level as "luck" to the LootContext so that it can be used
+        // TODO Properly introduce EntityPlayer as new param of getGrassSeed
+        LootContext.Builder contextBuilder = new LootContext.Builder(FMLCommonHandler.instance().getMinecraftServerInstance().getWorld(0)).withLuck(fortune);
```

Comment:
Please store `FMLCommonHandler.instance().getMinecraftServerInstance().getWorld(0)` to a local variable, it is used two times in this method.